openlibrary/catalog/marc/parse.py

```diff
@@ -337,18 +339,17 @@ def read_languages(rec: MarcBase, lang_008: str | None = None) -> list[str]:
     for f in rec.get_fields('041'):
         if f.ind2() == '7':
             code_source = ' '.join(f.get_subfield_values('2'))
-            # TODO: What's the best way to handle these?
-            raise MarcException("Non-MARC language code(s), source = ", code_source)
-            continue  # Skip anything which is using a non-MARC code source e.g. iso639-1
```

Comment:
Thanks for adding the logging. I should have noticed this before, but the `raise` + `continue` before doesn't make any sense, but I think you still want the `continue` in the new code so that it skips non-MARC codes (the comment is also useful to hint at what `ind2 == 7` means.

Reply:
Done. I think I had a half-idea that maybe the iso code might possibly be recognised correctly as a MARC code, so it was worth trying, but that seems quite hypothetical. I have not encountered a binary MARC with a $7 field yet, and is probably just as likely to result in a false allocation.

I agree that the comment is helpful, and this is more honest; logging that we don't recognise language codes we don't recognise, and moving on, without blocking the overall import.  If we log many other code sources, we can decide to support the common ones.